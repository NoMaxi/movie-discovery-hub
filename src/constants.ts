import { Genre, SelectableGenre, SortOption } from "@/types/common";

export const PRIMARY_COLOR = "#F65261";

export const GENRES: Genre[] = ["All", "Documentary", "Comedy", "Horror", "Crime"];
export const SELECTABLE_GENRES: SelectableGenre[] = ["Documentary", "Comedy", "Horror", "Crime"];

export const DEFAULT_SORT_CRITERION: SortOption = "Release Date";
export const DEFAULT_ACTIVE_GENRE: Genre = "All";
export const DEFAULT_MOVIES_PER_PAGE = 18;
