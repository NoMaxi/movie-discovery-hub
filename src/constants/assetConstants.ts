import noPosterImage from "@/assets/no_image_poster.png";

export const DEFAULT_IMAGE_URL = noPosterImage;
