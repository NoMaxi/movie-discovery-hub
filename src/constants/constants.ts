import { Genre, SelectableGenre, SortOption } from "@/types/common";

export const GENRES: Genre[] = ["All", "Documentary", "Comedy", "Horror", "Crime"];
export const SELECTABLE_GENRES: SelectableGenre[] = ["Documentary", "Comedy", "Horror", "Crime"];

export const PRIMARY_COLOR = "#F65261";

export const DEFAULT_SORT_CRITERION: SortOption = "Release Date";
export const DEFAULT_ACTIVE_GENRE: Genre = "All";
export const DEFAULT_TITLE = "No Title";
export const DEFAULT_DESCRIPTION = "No description available.";
export const DEFAULT_MOVIES_PER_PAGE = 18;
export const MOVIE_LIST_STALE_TIME = 5 * 60 * 1000; // 5 minutes
