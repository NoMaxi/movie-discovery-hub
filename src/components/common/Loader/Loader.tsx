import { PacmanLoader } from "react-spinners";
import { PRIMARY_COLOR } from "@/constants/constants";

export const Loader = () => (
    <PacmanLoader
        className="-ml-24"
        color={PRIMARY_COLOR}
        loading={true}
        size={40}
        aria-label="Loading Spinner"
        data-testid="loader"
    />
);
