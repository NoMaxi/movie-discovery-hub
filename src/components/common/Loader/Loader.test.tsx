import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { PacmanLoader } from "react-spinners";
import { PRIMARY_COLOR } from "@/constants/constants";
import { Loader } from "./Loader";

jest.mock("react-spinners", () => ({
    PacmanLoader: jest.fn(() => <div data-testid="mocked-pacman-loader">Mocked PacmanLoader</div>),
}));

describe("Loader", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test("Should render loader component with correct props", () => {
        const { asFragment } = render(<Loader />);

        expect(screen.getByTestId("mocked-pacman-loader")).toBeInTheDocument();
        expect(PacmanLoader).toHaveBeenCalledWith(
            {
                color: PRIMARY_COLOR,
                loading: true,
                size: 40,
                "aria-label": "Loading Spinner",
            },
            undefined,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
