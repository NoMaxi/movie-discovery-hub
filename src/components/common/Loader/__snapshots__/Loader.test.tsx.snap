// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should render loader component with correct props 1`] = `
<DocumentFragment>
  <div
    data-testid="mocked-pacman-loader"
  >
    Mocked PacmanLoader
  </div>
</DocumentFragment>
`;
